yark/cli.py

```diff
@@ -205,35 +206,95 @@ def parse_maximum_int(config_arg: str) -> int:
         if len(args) > 1:
-            # Get name
-            archive_name = args[1]
-
-            # Jank archive check
-            if not Path(archive_name).exists():
-                _err_archive_not_found()
-
-            # Launch and start browser
-            print(f"Starting viewer for {archive_name}..")
-            webbrowser.open(f"http://127.0.0.1:7667/archive/{archive_name}/videos")
-            launch()
-
-        # Start on archive finder
-        else:
-            print("Starting viewer..")
-            webbrowser.open("http://127.0.0.1:7667/")
-            launch()
+
+            def parse_value(config_arg: str) -> str:
+                return config_arg.split("=")[1]
+
+            def parse_port_int(config_arg: str) -> int:
+                """Tries to parse a port number input"""
+                port = parse_value(config_arg)
+                try:
+                    portInt = int(port)
+                    if portInt < 1 or portInt > 65535:
+                        raise Exception("invalid port number")
+                    return portInt
+                except Exception:
+                    print(HELP, file=sys.stderr)
+                    _log_err(
+                        f"\nError: The value '{port}' isn't a valid port number"
+                    )
+                    sys.exit(1)
+
+            # Go through each configuration argument
+            for config_idx in range(1, len(args)):
+                config_arg = args[config_idx]
+
+                # Custom bind host
+                if config_arg.startswith("--host="):
+                    config.bind_host = parse_value(config_arg)
+
+                # Custom bind port
+                elif config_arg.startswith("--port="):
+                    config.bind_port = parse_port_int(config_arg)
+
+                # Disable open in webbrowser behaviour
+                elif config_arg.startswith("--disable-webbrowser"):
+                    config.open_in_webbrowser = False
+
+                # Unknown argument
+                else:
+                    # if config_idx is 1 (first parameter could be name of archive)
+                    if config_idx == 1:
+                        archive_name = config_arg
+                    else:
+                        print(HELP, file=sys.stderr)
+                        _log_err(
+                            f"\nError: Unknown configuration '{config_arg}' provided for archive view"
+                        )
+                        sys.exit(1)
+
+        # Submit config settings
+        config.submit()
+
+        if config.open_in_webbrowser:
+            if archive_name is not None:
+                try:
+                    if not Path(archive_name).exists():
+                        raise ArchiveNotFoundException("Archive doesn't exist, please make sure you typed it's name correctly!")
+
+                    url = config.browser_url(archive_name)
+
+                    print(f"Starting viewer for {url}..")
```

Comment:
The convention inside of the project is that archives are `path`, as `archive_name` is available here it's prefered

Reply:
I persisted `archive_name` into the config functions as it's not a full path at that point it's just the archive name, should I change those ?

Reply:
Oh actually if you replace "for" to "at" having the url on the viewer is nice. Two ideal options:

1. `Starting viewer for [name] archive..` then `Viewer open at [url]`
2. `Starting viewer..` then `Viewer started at [url]`

Not sure if flask pops up with the url automatically

Reply:
How does this look.

`headless=False`, `archive_name=foobar`
```shell
vscode ➜ /tmp $ yark view foobar
Opening viewer at http://127.0.0.1:7667/archive/foobar/videos..
Starting server at http://127.0.0.1:7667/archive/foobar/videos for foobar
 * Serving Flask app 'yark.viewer'
 * Debug mode: off
```

`headless=False`, `archive_name=`
```shell
vscode ➜ /tmp $ yark view
Opening viewer at http://127.0.0.1:7667/..
Starting server at http://127.0.0.1:7667/
 * Serving Flask app 'yark.viewer'
 * Debug mode: off
```

`headless=True`, `archive_name=foobar`
```shell
vscode ➜ /tmp $ yark view foobar --headless
Starting server at http://127.0.0.1:7667/archive/foobar/videos for foobar
 * Serving Flask app 'yark.viewer'
 * Debug mode: off
```

`headless=True`, `archive_name=`
```shell
vscode ➜ /tmp $ yark view --headless
Starting server at http://127.0.0.1:7667/
 * Serving Flask app 'yark.viewer'
 * Debug mode: off
```

Reply:
Hmm maybe simplify it with a single line: `Starting server at [url][suffix]` with the suffix being ` for [name]` if name is present then it'll be good to go.

Reply:
It'd probably be better to just put spaces between the [url] and the ellipsis.
eg. `Starting server at [url][suffix] ...` and then remove the "Opening" for the webbrowser.

Where suffix can be empty or ` for [name]`

Examples:

```shell
vscode ➜ /tmp $ yark view foobar
Starting server at http://127.0.0.1:7667/archive/foobar/videos for foobar ...
 * Serving Flask app 'yark.viewer'
 * Debug mode: off
vscode ➜ /tmp $ yark view
Starting server at http://127.0.0.1:7667/ ...
 * Serving Flask app 'yark.viewer'
 * Debug mode: off
vscode ➜ /tmp $ yark view foobar --headless
Starting server at http://127.0.0.1:7667/archive/foobar/videos for foobar ...
 * Serving Flask app 'yark.viewer'
 * Debug mode: off
vscode ➜ /tmp $ yark view --headless
Starting server at http://127.0.0.1:7667/ ...
 * Serving Flask app 'yark.viewer'
 * Debug mode: off
```

Reply:
The styling convention which I like is `x..` instead of `x ...`

Python-y pseudocode for the address one:

```python
def loading_msg(url:str, archive_name:str|None=None):
    msg = f"Starting archive at {url} address"
    if archive_name is not None:
        msg += f" for {archive_name} archive"
    return f"{msg}.."
```

Reply:
Updated, put the `loading_msg` into the `launch` define.